src/ServiceControl.Audit.Persistence.Tests.RavenDB/IndexSetupTests.cs

```diff
@@ -0,0 +1,58 @@
+namespace ServiceControl.Audit.Persistence.Tests;
+
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using Persistence.RavenDB;
+using Persistence.RavenDB.Indexes;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Operations.Indexes;
+
+[TestFixture]
+class IndexSetupTests : PersistenceTestFixture
+{
+    [Test]
+    public async Task Corax_should_be_the_default_search_engine_type()
+    {
+        var indexes = await configuration.DocumentStore.Maintenance.SendAsync(new GetIndexesOperation(0, int.MaxValue));
+
+        foreach (var index in indexes)
+        {
+            var indexStats = await configuration.DocumentStore.Maintenance.SendAsync(new GetIndexStatisticsOperation(DatabaseSetup.MessagesViewIndexWithFulltextSearchName));
+            Assert.That(indexStats.SearchEngineType, Is.EqualTo(SearchEngineType.Corax), $"{index.Name} is not using Corax");
+        }
+    }
+
+    [Test]
+    public async Task Free_text_search_index_should_be_used_by_default()
+    {
+        var freeTextIndex = await configuration.DocumentStore.Maintenance.SendAsync(new GetIndexOperation(DatabaseSetup.MessagesViewIndexWithFulltextSearchName));
+        var nonFreeTextIndex = await configuration.DocumentStore.Maintenance.SendAsync(new GetIndexOperation(DatabaseSetup.MessagesViewIndexName));
+
+        Assert.That(nonFreeTextIndex, Is.Null);
+        Assert.That(freeTextIndex, Is.Not.Null);
+    }
+
+    [Test]
+    public async Task Indexes_should_be_reset_on_setup()
+    {
+        var index = new MessagesViewIndexWithFullTextSearch { Configuration = { ["Indexing.Static.SearchEngineType"] = SearchEngineType.Lucene.ToString() } };
+
+        await IndexCreation.CreateIndexesAsync([index], configuration.DocumentStore);
+
+        //TODO: find a better way
+        await Task.Delay(1000);
```

Comment:
@mauroservienti do you know of a way to wait for the Raven to complete the operations to avoid this hack?